plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -420,6 +402,28 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                 deletePredicate);
     }
 
+    private DeleteManager getDeleteManager(PartitionSpec partitionSpec, PartitionData partitionData)
+    {
+        if (partitionSpec.isUnpartitioned()) {
+            return unpartitionedTableDeleteManager;
+        }
+
+        Types.StructType structType = partitionSpec.partitionType();
+        PartitionKey partitionKey = partitionKeyFactories.computeIfAbsent(
```

Comment:
`PartitionSpec` and `PartitionData` will now be retained by provider. `PartitionData` particularly contains all partition values. How big are these structures when retained? In the past we had issues while retaining Hdfs classes: https://github.com/trinodb/trino/pull/14408

Do you think this could lead to memory issues with high concurrency and a lot of partitions? Let's say you run 30 queries at a time, each task querying 100k partitions with date field. If we get 1k bytes per partition with all objects overhead, then it's 3GB of unaccounted mem on a worker. If overhead is like 100bytes, then it's less of an issue.

Reply:
Yes.  This should have memory accounting added at some point.  The old code didn't have memory accounting either, and used significantly more memory.  I also think we need to move this to a flat hash as it is more efficient and easier to meaure for accounting.